allensdk/api/cloud_cache/README.md

```diff
@@ -0,0 +1,101 @@
+
+`CloudCache.metadata_path(metadata_fname)` will download a metadata file to
+the local system and return the path where the file has been stored. The list
+of valid values for `metadata_fname` can be found with
+`CloudCache.metadata_file_names`. If users wish to directly access a
+pandas DataFrame of a given metadata file, they can use
+`CloudCache.metadata(metadata_fname)`.
+
+## Structure of `manifest.json`
+
+The `manifest.json` files are structured like so
+```
+
+{
+ "dataset_version" : dataset_version_string,
+ "file_id_column": name_of_column_uniquely_identifying_files,
+ "metadata_files":{
+     metadata_file_name_1: {"uri": "full/uri/to/file",
+                            "version_id": version_id_string,
+                            "file_hash": file_hash_of_metadata_file},
+     metadata_file_name_2: {"uri": "full/uri/to/file",
+                            "version_id": version_id_string,
+                            "file_hash": file_hash_of_metadata_file},
+  ...
+ },
+ "data_files": {
+     file_id_1: {"uri": "full/uri/to/imaging_plane.nwb",
+                 "version_id": version_id_string,
+                 "file_hash": file_hash_of_file},
+     file_id_2: {"uri": "full/uri/to/behavior_only_session.nwb",
+                 "version_id": version_id_string,
+                 "file_hash": file_hash_of_file},
+    ...
+    }
+}
+```
+The entries under `metadata_files` and `data_files` provide the information
+necessary for `CloudCache` to
+
+- locate the online resoure
+- determine where it should be stored locally
+- determine if the copy that is stored locally is valid
+
+When a user asks to download a file, `CloudCache._manifest` (an instantiation
+of the `Manifest` class defined in `manifest.py`) constructs a candidate local
+path for the resource like
+```
+cache_dir/file_hash/relative_path_to_resource
+```
+where `cache_dir` is a parent directory for all local data storage specified by
+the user upon instantiated `CloudCache`. If a file already exists at that
+location, `CloudCache` compares its `file_hash` to the `file_hash` reported in
+the manifest. If they match, the file does not need to be downloaded. If either
+
+- a file does not exist at the candidate local path or
+- the `file_hash` of the file at the candidate local path does not match the
+`file_hash` reported in the manifest
+
+then `CloudCache` downloads the online resource to the candidate local path.
+By including `file_hash` in the local path, we ensure that, if `data_file_1`
+did not change between versions 1 and 2 of the dataset, it will not be
+needlessly downloaded again when the user switches between those versions of
+the dataset. Furthermore, when the user switches to version 3 of the dataset,
+they will not lose the old version of `data_file_1` that they previously
+downloaded, the `CloudCache` will merely redirect them to using the newer
+version of the data file.
+
+The `version_id` entry in the `manifest.json` descriptoin if resources is
```

Comment:
I think you meant to not write "descriptoin if resources"

Reply:
Fixed